presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/MergeJoinOptimizer.java

```diff
@@ -53,7 +55,7 @@ public PlanNode optimize(PlanNode plan, Session session, TypeProvider type, Plan
         requireNonNull(variableAllocator, "variableAllocator is null");
         requireNonNull(idAllocator, "idAllocator is null");
 
-        if (preferMergeJoin(session)) {
+        if (preferMergeJoin(session) && isGroupedExecutionEnabled(session)) {
```

Comment:
The GroupedExecutionEnabled session property is just a hint right? Are there cases where it is set to true but group execution is not enabled?